create Calendar Day component

skaik-mo opened the pull request:
### Preview
<img width="451" alt="days component" src="https://github.com/user-attachments/assets/66d9da71-5f80-4012-9d03-0c52cc6033cb" />

### Example Usage
```
    var isSelected by remember { mutableStateOf(true) }
    TudeeTheme(isDarkTheme = false) {
        CalendarDay(
            date = "15",
            day = "Mon",
            isSelected = isSelected,
            onClick = {
                isSelected = !isSelected
            }
        )
    }
```

skaik-mo committed:
create Calendar Day component

Belal-Alkhatib commented on app/src/main/java/com/qudus/tudee/ui/composable/CalendarDay.kt:
```diff
@@ -0,0 +1,123 @@
+    date: String,
+    day: String,
+    isSelected: Boolean = false,
+    onClick: () -> Unit,
```
It’s generally a good approach to put the content lambda as the last parameter in a composable function to support the trailing lambda style. But in this case, placing onClick at the end might be confusing — it could look like the caller needs to pass a composable instead of just a click action. So I’d suggest keeping onClick earlier in the parameter list to avoid that misunderstanding.

Belal-Alkhatib commented on app/src/main/java/com/qudus/tudee/ui/composable/CalendarDay.kt:
```diff
@@ -0,0 +1,123 @@
+
+@Composable
+fun CalendarDay(
+    modifier: Modifier = Modifier,
```
Good to place the modifier as the first optional parameter after required ones — it follows Compose conventions and improves usability.

skaik-mo committed:
Reorder CalendarDay parameters

mohammed-akkad commented on app/src/main/java/com/qudus/tudee/ui/composable/CalendarDay.kt:
```diff
@@ -0,0 +1,123 @@
+        modifier = modifier
+            .clip(RoundedCornerShape(16.dp))
+            .then(background)
+            .padding(vertical = 12.dp, horizontal = 12.dp)
```
You can apply padding using padding(12.dp) since it applies the same 12 dp value to both vertical and horizontal directions.

skaik-mo committed:
Refactor: Use uniform padding for CalendarDay

The padding for the CalendarDay composable was updated to use a uniform value of 12.dp instead of separate vertical and horizontal padding.

skaik-mo committed:
feat: Added PreviewLightDark to CalendarDay composable

This commit adds the `@PreviewLightDark` annotation to the `CalendarDay` composable preview, allowing it to be viewed in both light and dark themes.

The following previews were removed as they are now covered by the new annotation:
- `CalendarDayLightUnselectedPreview`
- `CalendarDayDarkSelectedPreview`
- `CalendarDayDarkUnselectedPreview`
